# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/ContentCompositorBridgeParent.h
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 gfx/layers/ipc/ContentCompositorBridgeParent.h
--- a/gfx/layers/ipc/ContentCompositorBridgeParent.h	Wed Jan 08 22:19:14 2020 +0000
+++ b/gfx/layers/ipc/ContentCompositorBridgeParent.h	Wed Jan 08 22:19:16 2020 +0000
@@ -161,18 +161,8 @@
     return IPC_FAIL_NO_REASON(this);
   }
 
-  dom::PWebGLParent* AllocPWebGLParent(
-      const WebGLVersion& aVersion, const HostWebGLCommandSink& aCommandSink,
-      const HostWebGLErrorSource& aErrorSource) override {
-    MOZ_ASSERT_UNREACHABLE("Unused");
-    return nullptr;
-  }
-
-  dom::PWebGLParent* AllocPWebGLParent(
-      const WebGLVersion& aVersion, HostWebGLCommandSink&& aCommandSink,
-      HostWebGLErrorSource&& aErrorSource) override;
-
-  bool DeallocPWebGLParent(PWebGLParent* aWebGLParent) override;
+  already_AddRefed<dom::PWebGLParent> AllocPWebGLParent(
+      const webgl::InitContextDesc&, webgl::InitContextResult* out) override;
 
   // Use DidCompositeLocked if you already hold a lock on
   // sIndirectLayerTreesLock; Otherwise use DidComposite, which would request